aten/src/ATen/cuda/CUDAGeneratorImpl.cpp

```diff
@@ -219,44 +334,52 @@ void CUDAGeneratorImpl::set_state(const c10::TensorImpl& new_state) {
   this->set_philox_offset_per_thread(static_cast<uint64_t>(philox_offset));
 }
 
+/**
+ * Sets the generator's current state to
+ * This function allows switching between different registered states of
+ * the generator.
+ */
+void CUDAGeneratorImpl::graphsafe_set_state(
+    const c10::intrusive_ptr<GeneratorImpl>& gen) {
+  c10::intrusive_ptr<CUDAGeneratorImpl> cuda_gen =
+      dynamic_intrusive_pointer_cast<CUDAGeneratorImpl>(gen);
+  TORCH_CHECK(cuda_gen, "Expected a CUDA Generator");
+  state_ = cuda_gen->state_;
```

Comment:
std::move here

Reply:
I intend to preserve `state_` within `cuda_gen`. This approach allows `cuda_gen` to continue referencing its state, enabling us to set this state in other graphs using the same generator.